File: api/prototipos.py
"""
Endpoint: /api/prototipos
Obtención de información de prototipos
"""
from http.server import BaseHTTPRequestHandler
import json
import logging
from lib.agente import AgenteInmobiliario

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        """Obtiene lista de prototipos disponibles"""
        try:
            # Obtener prototipos
            agente = AgenteInmobiliario()
            prototipos = agente.obtener_prototipos()

            # Responder
            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(json.dumps({
                'total': len(prototipos),
                'prototipos': prototipos
            }, ensure_ascii=False).encode('utf-8'))

        except Exception as e:
            logger.error(f"Error en /api/prototipos: {e}")
            self.send_response(500)
            self.send_header('Content-type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(json.dumps({
                'error': f'Error obteniendo prototipos: {str(e)}'
            }).encode())

    def do_OPTIONS(self):
        """Manejo de preflight CORS"""
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()

File: api/prototipos.js
/**
 * Vercel Serverless Function: /api/prototipos
 * Returns available property prototypes
 * Replaces the Python version (prototipos.py)
 */

import { readFileSync } from 'fs';
import { join } from 'path';

export default function handler(req, res) {
  res.setHeader('Content-Type', 'application/json');
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  if (req.method !== 'GET') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  try {
    const filePath = join(process.cwd(), 'data', 'prototipos.json');
    const data = JSON.parse(readFileSync(filePath, 'utf-8'));
    const prototipos = data.prototipos || [];

    return res.status(200).json({
      total: prototipos.length,
      prototipos
    });
  } catch (error) {
    console.error('Error loading prototipos:', error);
    return res.status(500).json({
      error: `Error obteniendo prototipos: ${error.message}`
    });
  }
}
